src/hooks/reducer.ts

```diff
@@ -1,9 +1,9 @@
-import { deepEqual } from "../utils/deepEqual";
+import {deepEqual} from '../utils/deepEqual';
 
 /**
  * @ignore
  */
-const reducer = (state, action) => {
+const reducer = (state: any, action: any) => {
```

Comment:
We should be able to avoid any here, for example like how it's done with auth0-react: https://github.com/auth0/auth0-react/blob/master/src/reducer.tsx#L20